amethyst_network/src/bundle.rs

```diff
@@ -0,0 +1,69 @@
+use super::NetSocketSystem;
+use amethyst_core::bundle::{Result, SystemBundle};
+use amethyst_core::shred::DispatcherBuilder;
+use filter::NetFilter;
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+use std::net::SocketAddr;
+
+/// A convenience bundle to create the infrastructure needed to send and receive network messages.
+pub struct NetworkBundle<'a, T> {
+    /// The local ip to bind to.
+    ip: &'a str,
+    /// The local port to bind to.
+    port: Option<u16>,
+    /// The filters applied on received network events.
+    filters: Vec<Box<NetFilter<T>>>,
+    /// The server to automatically connect to.
+    /// You would usually want this if you set is_server = false.
+    connect_to: Option<SocketAddr>,
+}
+
+impl<'a, T> NetworkBundle<'a, T> {
+    /// Creates a new NetworkBundle in client mode.
+    pub fn new_client(ip: &'a str, port: Option<u16>, filters: Vec<Box<NetFilter<T>>>) -> Self {
```

Comment:
You may want to forbid ports < 1024 here, as those require root. Or do a permissions check to tell the user they need to sudo.

Reply:
nice catch

Reply:
I added a warning here just in case someone is crazy enough to use it.